x/evm/types/msg_deposit.go

```diff
@@ -38,6 +44,14 @@ func (m ConfirmDepositRequest) ValidateBasic() error {
 		return sdkerrors.Wrap(err, "invalid chain")
 	}
 
+	if m.Amount.Equal(sdk.ZeroUint()) {
+		return fmt.Errorf("amount cannot be equal to 0")
+	}
+
+	if m.Amount.GT(sdk.NewUint(maxInt64)) {
```

Comment:
ERC20's amount is of type `uint256` in the contracts. The amount can totally go way beyond `int64`.

Reply:
I see. Allowing `uint256` will affect `vald` too. I'll have to take a look.

Reply:
Fixing this in another PR